shell/platform/windows/flutter_window_win32_unittests.cc

```diff
@@ -384,6 +388,59 @@ TEST(FlutterWindowWin32Test, NonPrintableKeyDownPropagation) {
   }
 }
 
+// Tests key event propagation of system (WM_SYSKEYDOWN) key down events.
+TEST(FlutterWindowWin32Test, SystemKeyDownPropagation) {
+  ::testing::InSequence in_sequence;
+
+  constexpr WPARAM virtual_key = VK_LEFT;
+  constexpr WPARAM scan_code = 10;
+  constexpr char32_t character = 0;
+  MockFlutterWindowWin32 win32window;
+  auto window_binding_handler =
+      std::make_unique<::testing::NiceMock<MockWindowBindingHandler>>();
+  TestFlutterWindowsView flutter_windows_view(
+      std::move(window_binding_handler), virtual_key, false /* is_printable */,
+      true /* is_syskey */);
+  win32window.SetView(&flutter_windows_view);
+  LPARAM lparam = CreateKeyEventLparam(scan_code, false, false);
+
+  // Test an event not handled by the framework
+  {
+    test_response = false;
+    flutter_windows_view.SetEngine(std::move(GetTestEngine()));
+    EXPECT_CALL(*flutter_windows_view.key_event_handler,
+                KeyboardHook(_, virtual_key, scan_code, WM_SYSKEYDOWN,
+                             character, false /* extended */, _))
+        .Times(2)
+        .RetiresOnSaturation();
+    EXPECT_CALL(*flutter_windows_view.text_input_plugin,
+                KeyboardHook(_, _, _, _, _, _, _))
+        .Times(1)
```

Comment:
Are you sure this will work? Is it based on actual tests? Because I don't think `SendInput` can send `SYS` key events, therefore the redispatching process, which `text_input_plugin` relies on, will not work correctly.